lib/Persistence/AbstractPersistence.php

```diff
@@ -88,16 +88,18 @@ protected static function _initialize()
                 throw new Exception('unable to create directory ' . self::$_path, 10);
             }
         }
-        $file = self::$_path . DIRECTORY_SEPARATOR . '.htaccess';
-        if (!is_file($file)) {
-            $writtenBytes = @file_put_contents(
-                $file,
-                'Require all denied' . PHP_EOL,
-                LOCK_EX
-            );
-            if ($writtenBytes === false || $writtenBytes < 19) {
-                throw new Exception('unable to write to file ' . $file, 11);
-            }
+        if ((strncmp("Apache", $_SERVER['SERVER_SOFTWARE'],6) == 0) or defined('_SERVER_APACHE')) {
```

Comment:
The unit tests fail due to the missing check if the key SERVER_SOFTWARE exists. Also we need to add tests that mocks the existence of that variable emulating both an apache and non-apache environments to ensure the .htaccess files get created in Apache and not in others.

I can look into writing these tests later some time during this week.